Source/WebCore/platform/graphics/wayland/PlatformDisplayWayland.cpp

```diff
@@ -122,6 +123,43 @@ void PlatformDisplayWayland::initialize()
     m_registry.reset(wl_display_get_registry(m_display));
     wl_registry_add_listener(m_registry.get(), &s_registryListener, this);
     wl_display_roundtrip(m_display);
+}
+
+#if PLATFORM(GTK)
+EGLDisplay PlatformDisplayWayland::gtkEGLDisplay()
+{
+    if (m_eglDisplay != EGL_NO_DISPLAY)
+        return m_eglDisplayOwned ? EGL_NO_DISPLAY : m_eglDisplay;
+
+    if (!m_sharedDisplay)
+        return EGL_NO_DISPLAY;
+
+#if USE(GTK4)
+    m_eglDisplay = gdk_wayland_display_get_egl_display(m_sharedDisplay.get());
+#else
+    auto* window = gtk_window_new(GTK_WINDOW_POPUP);
+    gtk_widget_realize(window);
+    if (auto context = adoptGRef(gdk_window_create_gl_context(gtk_widget_get_window(window), nullptr))) {
+        gdk_gl_context_make_current(context.get());
+        m_eglDisplay = eglGetCurrentDisplay();
+    }
+#endif
```

Comment:
This leaks the window returned by `gtk_window_new()`, it would be better to adopt the pointer into a `GRefPtr`:

 ```suggestion
    auto window = adoptGRef(gtk_window_new(GTK_WINDOW_POPUP));
    gtk_widget_realize(window.get());
    if (auto context = adoptGRef(gdk_window_create_gl_context(gtk_widget_get_window(window.get()), nullptr))) {
        gdk_gl_context_make_current(context.get());
        m_eglDisplay = eglGetCurrentDisplay();
    }
#endif
```

Reply:
Right, I forgot the destroy, but widgets are initially unowned so you can't adopt the floating ref. I'll just use destroy. Thanks!

Reply:
Adopting uses `g_object_ref_sink()`, a smart pointer would work. If you prefer to use `gtk_widget_destroy()` that's fine as well.

Reply:
No, adopting doesn't do any ref. For widgets it's preferred to use destroy.